Sources/NIOCore/CircularBuffer.swift

```diff
@@ -251,6 +251,10 @@ extension CircularBuffer: Collection, MutableCollection {
 
         return (self[self.endIndex..<self.endIndex].makeIterator(), self.count)
     }
+    
+    public func _failEarlyRangeCheck(_ index: Index, bounds: Range<Index>) {}
+    public func _failEarlyRangeCheck(_ index: Index, bounds: ClosedRange<Index>) {}
+    public func _failEarlyRangeCheck(_ range: Range<Index>, bounds: Range<Index>) {}
```

Comment:
Can we mark these all `@inlinable`, and also note that these have empty bodies for performance reasons?

Reply:
Fantastic, now can we add a comment above these saying that they're empty for performance reasons?

Reply:
@Lukasa ok I added the comment in [135f5e7](https://github.com/apple/swift-nio/pull/2161/commits/135f5e744c99ce46070b45235014a6f89dbf2971).

Just for my understanding, now as these are implemented as no-ops, **there is no range check on index range access at runtime** (so we don't have these extra calls at runtime)?

The default behavior was:
> The default implementation for random access indices performs a range check.

If so, where were these methods called from? I was under the impression that these are supposed to be called by us not Swift?

Reply:
No, these are called by Swift. Swift calls them from algorithm implementations on Swift. We're not supposed to call them.

Both `CircularBuffer` and `Array` do their own checking so we don't need to have Swift do it here.

Reply:
> No, these are called by Swift. Swift calls them from algorithm implementations on Swift. We're not supposed to call them.
> 
> Both `CircularBuffer` and `Array` do their own checking so we don't need to have Swift do it here.

Ah, ok that is what I was wondering about, thanks!